Fix a few small bugs and one big memory leak.

ubermichael opened the pull request:
This patch fixes a null reference error where the code assumes a content provider has a plugin. That might not be the case.

It also fixes a memory leak in the SOAP client.

ubermichael committed:
don't assume providers have plugins.

ubermichael committed:
switch to using an iterator for checking depsit status.

ubermichael committed:
Merge branch 'master' of github.com:ubermichael/lockss-o-matic

ubermichael committed:
Track down and fix the damn memory leak.

 * lom:deposit:status will now display memory usage along with the
progress if passed the -vv option.

 * Clean up the SoapClient in LockssSoapClient. This should happen
automatically whenever the client is replaced or falls out of scope. But
there must be some sort of referential loop in the client that prevents
it from being picked up by the garbage collector. So manually remote it.

ubermichael committed:
Merge branch 'master' of github.com:mjordan/lockss-o-matic

mjordan commented:
Looks good to me. Nice work on tracking down that memory leak.

Do we need to support PHP 5.4 and 5.5?

mjordan commented:
@ubermichael any idea why the remote in the last commit is still being referred to as `mjordan`?

ubermichael commented:
Probably because I haven't changed the remote URL yet. Looks like Github puts redirects in place for repositories that are transferred. Nifty!

ubermichael committed:
cleaning up a bit.

ubermichael committed:
use only active boxes when checking deposit status.

ubermichael committed:
Add date checked to deposit.

Select which deposits to check status based on least-recently
checked. make sure deposits are checked at most once a day.
